plugins/modules/monitoring/spectrum_model_attrs.py

```diff
@@ -0,0 +1,526 @@
+     - URL of OneClick server.
+     type: str
+     required: True
+   url_username:
+     description:
+     - OneClick username.
+     type: str
+     required: True
+     aliases: [username]
+   url_password:
+     description:
+     - OneClick password.
+     type: str
+     required: True
+     aliases: [password]
+   use_proxy:
+     description:
+     - if C(no), it will not use a proxy, even if one is defined in
+       an environment variable on the target hosts.
+     default: yes
+     required: False
+     type: bool
+   name:
+     description:
+     - Model name.
+     type: str
+     required: True
+   type:
+     description:
+     - Model type.
+     type: str
+     required: True
+   validate_certs:
+     description:
+     - Validate SSL certificates.
+     type: bool
+     default: yes
+     required: False
+   attributes:
+     description:
+     - A list of attribute names and values to enforce.
+     - All values and parameters are case sensitive and must be provided as strings only.
+     required: True
+     type: list
+     elements: dict
+     suboptions:
+       name:
+         description:
+         - Attribute name OR hex ID.
+         - Not all names are defined, see C(attr_map) var declaration in __init__() for a complete listing.
+         - Hex IDs are the direct identifiers in Spectrum and will always work.
+         - To lookup hex IDs go to the UI..
+         - Locator -> Devices -> By Model Name -> <enter any model> -> Attributes tab.
+         type: str
+         required: True
+       value:
+         description:
+         - Attribute value. Empty strings should be C("") or C(null).
+         type: str
+         required: True
+'''
+
+EXAMPLES = r'''
+- name: Enforce maintenance mode for modelxzy01 with a note about why
+  spectrum_model_attrs:
+    url: "http://oneclick.url.com"
+    username: "{{ oneclick_username }}"
+    password: "{{ oneclick_password }}"
+    name: "modelxyz01"
+    type: "Host_Device"
+    validate_certs: no
+    attributes:
+      - name: "isManaged"
+        value: "false"
+      - name: "Notes"
+        value: "MM set on {{ ansible_date_time.iso8601 }} via CO {{ CO }} by {{ tower_user_name | default(ansible_user_id) }}"
+  delegate_to: localhost
+  register: spectrum_model_attrs_status
+'''
+
+RETURN = r'''
+spectrum_model_attrs_status:
+    description: Information about changed model attributes.
+    returned: always
+    type: dict
+    sample: {
+        "msg": "Success",
+        "changed_attrs": {
+            "Notes": "MM set on 2021-02-03T22:04:02Z via CO CO9999 by tgates",
+            "isManaged": "true"
+        },
+    }
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible.module_utils.urls import fetch_url
+import json
+import re
+import xml.etree.ElementTree as ET
+try:
+    # python3
+    import urllib.parse as urlparse
+except ImportError:
+    # python2
+    import urlparse
```

Comment:
Please use `from ansible.module_utils.six.moves.urllib.parse import urlparse` instead of this construct.

Reply:
The purpose was to use it to urlencode a string but I can't seem to get it to work. Could you suggest?
current method:
`urlparse.quote(string, "<>%-_.!*'():?#/@&+,;=")`

six moves:
`>>> urlparse("afas asdfasdf  asdfasd")
ParseResult(scheme='', netloc='', path='afas asdfasdf  asdfasd', params='', query='', fragment='')
>>> 
>>> urlparse("https:/wwww.something.com/model?asfds&id=0x0015&val=afas asdfasdf", allow_fragments=True)
ParseResult(scheme='https', netloc='', path='/wwww.something.com/model', params='', query='asfds&id=0x0015&val=afas asdfasdf', fragment='')`

Reply:
If you need `quote`, then import it instead: `from ansible.module_utils.six.moves.urllib.parse import quote`

Reply:
commit #1802